app/wasm/test/custom_msg_test.go

```diff
@@ -14,46 +15,50 @@ import (
 	wasmbindings "github.com/osmosis-labs/osmosis/v7/app/wasm/bindings"
 )
 
-// func TestMintMsg(t *testing.T) {
-// 	creator := RandomAccountAddress()
-// 	osmosis, ctx := SetupCustomApp(t, creator)
+func TestMintMsg(t *testing.T) {
```

Comment:
This is a good sanity check of the happy path.

I would add something else to check for metadata, and also mint a few times from the same contract, With same and different denoms (eg. A, A, B) to ensure metadata is set properly and in all cases. (esp. that the "only create once" guard is correct)

Reply:
Will do.